### PR TITLE
BUG#PWA-3065 When the button sign in is pressed by the keyboard, noth…

### PR DESCRIPTION
**Pull request by RaghavendraTirumalasetti**
…ing happens
## 
Pre-Conditions:

1. Have Magento 2.4.6alpha2 EE setup with [Venia configured](https://wiki.corp.adobe.com/display/ACRQ/PWA+Studio+Set+Up+in+Local).
2. Make sure Venia has at least one customer login account to login into the Venia front end.
3. Make sure Screen reader installed and enabled.

Steps to Reproduce:

1. Open Venia front end in any browser 
2. Navigate by using tab button sign in button > press enter to expand 
3. Fill the details to sign in and press key ENTER from being on sign in button

Expected result: Sign in button should respond with its functionality in the UI when the ENTER key is pressed.

Actual result: Sign in button is not working with the keyboard when the ENTER key is pressed

## Related Issue

Closes # https://jira.corp.adobe.com/browse/PWA-3065.

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/peregrine/lib/talons/SignIn/__tests__/useSignIn.spec.js [hunk 1]
```diff
@@ -131,6 +131,7 @@ test('returns correct shape', () => {
             "shouldRender": false,
```
Comment (anthoula) [inline packages/peregrine/lib/talons/SignIn/__tests__/useSignIn.spec.js:131]: Hi @RaghavendraTirumalasetti - Would you sign the Adobe CLA? [Details](https://github.com/magento/pwa-studio/pull/4038/checks?check_run_id=11466390104)

Please fill out Description & remove TODO's for the PR description [Details](https://github.com/magento/pwa-studio/pull/4038#issuecomment-1436958883)

Afterwards, please rerun builds & they should pass 👍

Comment (anthoula) [reply packages/peregrine/lib/talons/SignIn/__tests__/useSignIn.spec.js:131]: Code looks good, thank you

**Review by anthoula — APPROVED** [review]

**Comment by glo42707** [comment]
run cypress

**Comment by dpatil-magento** [comment]
Good to merge after Adobe CLA is signed.

**Pull request closed without merging**